src/video_canister/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use ic_cdk::export::candid::{CandidType, Deserialize, Principal,};
+use ic_cdk_macros::{update, query, init, pre_upgrade, post_upgrade};
+use ic_cdk::storage;
+
+#[derive(CandidType, Deserialize)]
+pub struct PutMetaInfo{
+    pub name: String,
+    pub description: String,
+    pub chunk_num: usize,
+}
+
+#[derive(CandidType, Deserialize)]
+pub struct MetaInfo{
+    pub name: String,
+    pub description: String,
+    pub chunk_num: usize,
+    pub version: usize,
+    pub owner: Principal,
+}
+
+impl Default for MetaInfo{
+    fn default() -> Self { 
+        MetaInfo{
+            name: String::new(),
+            description: String::new(),
+            chunk_num: 0,
+            version: CANISTER_VERSION,
+            owner: Principal::anonymous(),
+        }
+     }
+}
+
+#[derive(CandidType, Deserialize)]
+pub enum PutMetaInfoResponse{
+    #[serde(rename = "success")]
+    Success,
+
+    #[serde(rename = "missing_rights")]
+    MissingRights,
+}
+
+#[derive(CandidType, Deserialize)]
+pub enum PutChunkResponse{
+    #[serde(rename = "success")]
+    Success,
+
+    #[serde(rename = "missing_rights")]
+    MissingRights,
+
+    #[serde(rename = "out_of_bounds")]
+    OutOfBounds
+}
+
+#[derive(CandidType, Deserialize)]
+pub enum ChangeOwnerResponse{
+    #[serde(rename = "success")]
+    Success,
+
+    #[serde(rename = "missing_rights")]
+    MissingRights,
+}
+
+type Chunk = Vec<u8>;
+type Chunks = Vec<Chunk>;
+
+const CANISTER_VERSION: usize = 0usize;
+
+#[init]
+pub async fn init(owner: Principal){
+    storage::get_mut::<MetaInfo>().owner = owner;
+}
+
+#[update]
+pub async fn put_meta_info(new_meta_info: PutMetaInfo) -> PutMetaInfoResponse{
+    let mut old_meta_info = storage::get_mut::<MetaInfo>();
+    if old_meta_info.owner != ic_cdk::caller() {
+        return PutMetaInfoResponse::MissingRights;
+    } else {
+        old_meta_info.name = new_meta_info.name;
+        old_meta_info.description = new_meta_info.description;
+
+        //Resize array
+        if old_meta_info.chunk_num != new_meta_info.chunk_num{
+            let _old_chunks = std::mem::replace(storage::get_mut::<Chunks>(), vec![Vec::new(); new_meta_info.chunk_num]);
+            old_meta_info.chunk_num = new_meta_info.chunk_num;
+        }
+
+        return PutMetaInfoResponse::Success;
+    }
```

Comment:
ich würde in der put methode allgemein nochmal über die variablenbenennung von "old_meta_info" nachdenken. "Old" hätte mich im ersten moment erwarten lassen dass am Ende gelöscht und replaced wird. Tatsächlich werden ja nur die Werte von "Old" upgedatet, richtig? Dann fände ich "existing" oder so besser.

Reply:
fixed